server/webhook.go

```diff
@@ -131,11 +146,198 @@ func (s *Server) registerWebhookRoutes(g *echo.Group) {
 		}
 		return c.String(http.StatusOK, strings.Join(createdMessages, "\n"))
 	})
+
+	// id is the webhookEndpointID in repository
+	// This endpoint is generated and injected into GitHub action & GitLab CI during the VCS setup.
+	g.POST("/sql-review/:id", func(c echo.Context) error {
+		body, err := io.ReadAll(c.Request().Body)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Failed to read SQL review request").SetInternal(err)
+		}
+		log.Debug("SQL review request received for VCS project",
+			zap.String("webhook_endpoint_id", c.Param("id")),
+			zap.String("request", string(body)),
+		)
+
+		var request vcsSQLReviewRequest
+		if err := json.Unmarshal(body, &request); err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Malformed SQL review request").SetInternal(err)
+		}
+
+		filter := func(repo *api.Repository) (bool, error) {
+			return c.Request().Header.Get("X-SQL-Review-Token") == repo.WebhookSecretToken && strings.HasPrefix(repo.WebURL, request.WebURL), nil
+		}
+		ctx := c.Request().Context()
+		repositoryList, err := s.filterRepository(ctx, c.Param("id"), request.RepositoryID, filter)
+		if err != nil {
+			return err
+		}
+		if len(repositoryList) == 0 {
+			log.Debug("Empty handle repo list. Ignore this request.")
+			return c.JSON(http.StatusOK, &vcsSQLReviewResult{
+				Status:  advisor.Success,
+				Content: []string{},
+			})
+		}
+
+		repo := repositoryList[0]
+		prFiles, err := vcs.Get(repo.VCS.Type, vcs.ProviderConfig{}).ListPullRequestFile(
+			ctx,
+			common.OauthContext{
+				ClientID:     repo.VCS.ApplicationID,
+				ClientSecret: repo.VCS.Secret,
+				AccessToken:  repo.AccessToken,
+				RefreshToken: repo.RefreshToken,
+				Refresher:    s.refreshToken(ctx, repo.WebURL),
+			},
+			repo.VCS.InstanceURL,
+			request.RepositoryID,
+			request.PullRequestID,
+		)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to list pull request file").SetInternal(err)
+		}
+
+		distinctFileList := []vcs.DistinctFileItem{}
```

Comment:
I found we can build the `DistinctFileItem` list manually, then reuse the `groupFileInfoByRepo` and `filterRepository`. PTAL @d-bytebase

Reply:
This sounds good!